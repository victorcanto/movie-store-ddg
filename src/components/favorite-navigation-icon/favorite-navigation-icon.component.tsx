import React, { useState } from 'react';
import { Badge, Tooltip } from '@mui/material';
import FavoriteIcon from '@mui/icons-material/Favorite';
import {
	useAppSelector,
	useAppDispatch,
	selectFavoritedQuantity,
	setIsFavoritesOpen,
	selectIsFavoritesOpen,
	setIsCartOpen,
} from '../../store';

export function FavoriteNavigationIcon() {
	const dispatch = useAppDispatch();

	const favoritedQuantity = useAppSelector(selectFavoritedQuantity);
	const isFavoritesOpen = useAppSelector(selectIsFavoritesOpen);

	const toggleIsFavoritesOpen = (): void => {
		dispatch(setIsFavoritesOpen(!isFavoritesOpen));
		dispatch(setIsCartOpen(false));
	};

	return (
		<Tooltip
			title='Filmes favoritos'
			data-testid='favorite-navigation-icon'
		>
			<Badge
				badgeContent={favoritedQuantity}
				color='secondary'
				onClick={toggleIsFavoritesOpen}
			>
				<FavoriteIcon fontSize='large' />
			</Badge>
		</Tooltip>
	);
}
