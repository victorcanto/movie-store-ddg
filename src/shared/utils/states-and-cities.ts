import statesAndCities from '../data/estados-cidades.json';

export const getBrazilianStates = () =>
	statesAndCities.estados.map(({ nome }) => nome);

export const getBrazilianCitiesByState = (stateName: string) =>
	statesAndCities.estados.find(({ nome }) => nome === stateName)?.cidades;
