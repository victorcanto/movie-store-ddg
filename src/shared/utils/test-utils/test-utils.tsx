import React, { PropsWithChildren } from 'react';
import { render as rtlRender } from '@testing-library/react';
import type { RenderOptions } from '@testing-library/react';
import type { PreloadedState } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { setupStore, AppStore, RootState } from '../../../store';

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
	preloadedState?: PreloadedState<RootState>;
	store?: AppStore;
}

export function renderWithProviders(
	ui: React.ReactElement,
	{
		preloadedState = {},
		store = setupStore(preloadedState),
		...renderOptions
	}: ExtendedRenderOptions = {},
) {
	function Wrapper({ children }: PropsWithChildren<{}>): JSX.Element {
		return (
			<Provider store={store}>
				<BrowserRouter>{children}</BrowserRouter>
			</Provider>
		);
	}
	return { store, ...rtlRender(ui, { wrapper: Wrapper, ...renderOptions }) };
}

export * from '@testing-library/react';

export { renderWithProviders as render };
