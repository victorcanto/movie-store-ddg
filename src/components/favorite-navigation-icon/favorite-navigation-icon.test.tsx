import React from 'react';
import userEvent from '@testing-library/user-event';
import { render } from '../../shared/utils/test-utils';
import { FavoriteNavigationIcon } from './favorite-navigation-icon.component';

test('snapshot of the base state', () => {
	const { asFragment } = render(<FavoriteNavigationIcon />);
	expect(asFragment()).toMatchSnapshot();
});

test('snapshot of the hover', () => {
	const { getByTestId, asFragment } = render(<FavoriteNavigationIcon />);

	userEvent.hover(getByTestId(/favorite-navigation-icon/i));

	expect(asFragment()).toMatchSnapshot();
});
