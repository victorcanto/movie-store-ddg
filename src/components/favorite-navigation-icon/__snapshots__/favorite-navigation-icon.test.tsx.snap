// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot of the base state 1`] = `
<DocumentFragment>
  <span
    aria-label="Filmes favoritos"
    class="MuiBadge-root BaseBadge-root css-1c32n2y-MuiBadge-root"
    data-mui-internal-clone-element="true"
    data-testid="favorite-navigation-icon"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
      data-testid="FavoriteIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="m12 21.35-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
      />
    </svg>
    <span
      class="MuiBadge-badge MuiBadge-standard MuiBadge-invisible MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorSecondary BaseBadge-badge BaseBadge-invisible css-fu0map-MuiBadge-badge"
    >
      0
    </span>
  </span>
</DocumentFragment>
`;

exports[`snapshot of the hover 1`] = `
<DocumentFragment>
  <span
    aria-label="Filmes favoritos"
    class="MuiBadge-root BaseBadge-root css-1c32n2y-MuiBadge-root"
    data-mui-internal-clone-element="true"
    data-testid="favorite-navigation-icon"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
      data-testid="FavoriteIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="m12 21.35-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
      />
    </svg>
    <span
      class="MuiBadge-badge MuiBadge-standard MuiBadge-invisible MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorSecondary BaseBadge-badge BaseBadge-invisible css-fu0map-MuiBadge-badge"
    >
      0
    </span>
  </span>
</DocumentFragment>
`;
